yaml.qmd

```diff
@@ -0,0 +1,874 @@
+---
+title: "Quotes: \U0001F4E8"
+subtitle: 'Quotes: \U0001F4E8'
+---
+```
+
+Q: where does the `\U` go in `subtitle` go?
+YAML might be interpreting literally but Quarto/Pandoc might escape somewhere?
+
+### Long strings
+
+For strings that are longer than you'd comfortably put on a single line, or need explicit line breaks, you can use what YAML calls block scalars.
+
+There are two styles: literal and folded.
+
+Literal block scalars are indicated by a pipe (`|`), and line breaks are considered significant, whereas folded block scalars are indicated by a greater than symbol (`>`), and line breaks are collapsed to spaces.
+
+In both cases, start your string on a new line indented under the key:
+
+::: {layout-ncol="2"}
+``` {.yaml filename="literal.qmd"}
+---
+title: |
+  A title
+
+  spanning two lines
+---
+```
+
+``` {.yaml filename="folded.qmd"}
+---
+title: >
+  A title
+
+  collapsed into one line
+---
+```
+:::
+
+::: {#fig-block-strings layout-ncol="2"}
+![Rendered title using literal (`|`) block scalar](images/yaml-block-literal.png){#fig-block-strings-literal .border fig-alt="A screenshot of a title that reads 'A title' then a line break followed by 'spanning two lines'."}
+
+![Rendered title using folded (`>`) block scalar](images/yaml-block-folded.png){#fig-block-strings-folded .border fig-alt="A screenshot of a title that reads 'A title collapsed into one line' all on a single line."}
+:::
+
+Depending on where the string value is used, line breaks may or may not mean anything to Quarto.
+For instance, when interpreted as markdown, a single line break doesn't usually result in anything in the output.
+However, a double line break will be interpreted as a new paragraph.
+
+### Using `default`
+
+You'll come across examples where the value of a Quarto option is `default`, like when you are specifying multiple formats:
+
+``` {.yaml filename="document.qmd"}
+---
+format:
+  html: default
+  pdf: default
+---
+```
+
+The value `default` doesn't have a special YAML meaning--- it's something Quarto adds for certain options where the value is usually another mapping, but you just want the default values.
+
+## What happens when you get it wrong?
+
+When you render a project or document, one the first things that happens is the options are parsed and validated.
+When you make a mistake in setting an option there are a few possible outcomes:
+
+1.  You get an error message, and the error message clearly points out your mistake.
+
+2.  You get an error message, but it's not clear from that message what or where your mistake is.
+
+3.  You don't get an error message at all.
+
+It's hard to enumerate all the ways you can make a mistake, but we'll give a couple of examples for each of these outcomes in the hope it helps you track down yours.
+
+Some examples in this chapter will result in an error from `quarto render`---we've highlighted these examples with a red border (TODO: what will we do in PDF), like this:
```

Comment:
Could we do a red border and a red X icon/emoji or something for the div?

Reply:
Yeah, sounds good. I'll leave the TODO for now, because it's bound to need something special for LaTeX.